cmake/BundledAwsSdkCpp.cmake

```diff
@@ -120,71 +92,100 @@ function(use_bundled_libaws SOURCE_DIR BINARY_DIR)
+    add_dependencies(AWS::aws-c-io aws-sdk-cpp-external)
+    target_include_directories(AWS::aws-c-io INTERFACE ${LIBAWS_INCLUDE_DIR})
+    target_link_libraries(AWS::aws-c-io INTERFACE AWS::aws-c-common)
+
     add_library(AWS::aws-checksums STATIC IMPORTED)
     set_target_properties(AWS::aws-checksums PROPERTIES IMPORTED_LOCATION "${BINARY_DIR}/thirdparty/libaws-install/${CMAKE_INSTALL_LIBDIR}/${PREFIX}aws-checksums.${SUFFIX}")
-    add_dependencies(AWS::aws-checksums aws-checksums-external)
-    set_property(TARGET AWS::aws-checksums APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${LIBAWS_INCLUDE_DIR})
-    set_property(TARGET AWS::aws-checksums APPEND PROPERTY INTERFACE_LINK_LIBRARIES CURL::libcurl OpenSSL::Crypto OpenSSL::SSL ZLIB::ZLIB Threads::Threads)
-    if (APPLE)
-        set_property(TARGET AWS::aws-checksums APPEND PROPERTY INTERFACE_LINK_LIBRARIES "-framework CoreFoundation")
-    endif()
+    add_dependencies(AWS::aws-checksums aws-sdk-cpp-external)
+    target_include_directories(AWS::aws-checksums INTERFACE ${LIBAWS_INCLUDE_DIR})
 
     add_library(AWS::aws-c-event-stream STATIC IMPORTED)
     set_target_properties(AWS::aws-c-event-stream PROPERTIES IMPORTED_LOCATION "${BINARY_DIR}/thirdparty/libaws-install/${CMAKE_INSTALL_LIBDIR}/${PREFIX}aws-c-event-stream.${SUFFIX}")
-    add_dependencies(AWS::aws-c-event-stream aws-c-event-stream-external)
-    set_property(TARGET AWS::aws-c-event-stream APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${LIBAWS_INCLUDE_DIR})
-    set_property(TARGET AWS::aws-c-event-stream APPEND PROPERTY INTERFACE_LINK_LIBRARIES AWS::aws-c-common AWS::aws-checksums CURL::libcurl OpenSSL::Crypto OpenSSL::SSL ZLIB::ZLIB Threads::Threads)
-    if (APPLE)
-        set_property(TARGET AWS::aws-c-event-stream APPEND PROPERTY INTERFACE_LINK_LIBRARIES "-framework CoreFoundation")
-    endif()
+    add_dependencies(AWS::aws-c-event-stream aws-sdk-cpp-external)
+    target_include_directories(AWS::aws-c-event-stream INTERFACE ${LIBAWS_INCLUDE_DIR})
+    target_link_libraries(AWS::aws-c-event-stream INTERFACE AWS::aws-checksums AWS::aws-c-io)
+
+    add_library(AWS::aws-c-auth STATIC IMPORTED)
+    set_target_properties(AWS::aws-c-auth PROPERTIES IMPORTED_LOCATION "${BINARY_DIR}/thirdparty/libaws-install/${CMAKE_INSTALL_LIBDIR}/${PREFIX}aws-c-auth.${SUFFIX}")
+    add_dependencies(AWS::aws-c-auth aws-sdk-cpp-external)
+    target_include_directories(AWS::aws-c-auth INTERFACE ${LIBAWS_INCLUDE_DIR})
+
+    add_library(AWS::aws-c-s3 STATIC IMPORTED)
+    set_target_properties(AWS::aws-c-s3 PROPERTIES IMPORTED_LOCATION "${BINARY_DIR}/thirdparty/libaws-install/${CMAKE_INSTALL_LIBDIR}/${PREFIX}aws-c-s3.${SUFFIX}")
+    add_dependencies(AWS::aws-c-s3 aws-sdk-cpp-external)
+    target_include_directories(AWS::aws-c-s3 INTERFACE ${LIBAWS_INCLUDE_DIR})
+    target_link_libraries(AWS::aws-c-s3 INTERFACE AWS::aws-c-auth)
```

Comment:
Did you consider switching to s3-crt? https://github.com/aws/aws-sdk-cpp/wiki/Improving-S3-Throughput-with-AWS-SDK-for-CPP-v1.9#working-with-the-new-s3-crt-client

Reply:
I did not try it yet, but I will check it out, possibly in a separate PR if it proves to be good for us.

Reply:
I checked out the S3 CRT client and one problem I realized is that the new `Aws::S3Crt::ClientConfiguration` does not have all the features the previously used `Aws::Client::ClientConfiguration` has. We currently have properties in our S3 processors for proxy options and a communications timeout option that are missing from the new configuration. I'm not sure if these will be introduced later, I'll ask the developers about these features using the S3 CRT lib.

Reply:
According to the devs these features are currently missing and are planned to be implemented in the future, but no timeline is set for it yet.

Reply:
You really went the extra mile! Thanks for all the research. In this case we're staying on the old client.